url2b521cbda7_eobermuhlner_jhuge_tgz-pJ8-ch_obermuhlner_jhuge_example_MeasureHeapJ8/src/ch/obermuhlner/jhuge/converter/CompactConverter.java

```diff
@@ -0,0 +1,657 @@
+			catch (IOException exception) {
+				// ignore
+			}
+		}
+		
+		byte[] serializedData = super.serialize(source);
+		byte[] compactData = new byte[serializedData.length + 1];
+		compactData[0] = SERIALIZED;
+		System.arraycopy(serializedData, 0, compactData, 1, serializedData.length);
+		return compactData;
+	}
+	
+	@Override
+	public T deserialize(byte[] data) {
+		if (data == null || data.length == 0) {
+			return null;
+		}
+
+		byte type = data[0];
+		switch (type) {
+		case NULL: {
+			return null;
+		}
+		case BYTE: {
+			byte value = data[1];
+			@SuppressWarnings("unchecked")
+			T result = (T) Byte.valueOf(value);
+			return result;
+		}
+		case SHORT: {
+			short value = (short) (
+					((data[1] & 0xff) << 8) + 
+					(data[2] & 0xff));
+			@SuppressWarnings("unchecked")
+			T result = (T) Short.valueOf(value);
+			return result;
+		}
+		case INT: {
+			int value = (
+					((data[1] & 0xff) << 24) + 
+					((data[2] & 0xff) << 16) + 
+					((data[3] & 0xff) << 8) + 
+					(data[4] & 0xff));
+			@SuppressWarnings("unchecked")
+			T result = (T) Integer.valueOf(value);
+			return result;
+		}
+		case LONG: {
+			long value = (
+					((long) data[1] << 56) +
+					((long) (data[2] & 255) << 48) +
+					((long) (data[3] & 255) << 40) +
+					((long) (data[4] & 255) << 32) +
+					((long) (data[5] & 255) << 24) +
+					((long) (data[6] & 255) << 16) +
+					((long) (data[7] & 255) << 8) +
+					((long) (data[8] & 255) << 0));
+
+			@SuppressWarnings("unchecked")
+			T result = (T) Long.valueOf(value);
+			return result;
+		}
+		case CHAR: {
+			char value = (char) (
+					((data[1] & 0xff) << 8) +
+					(data[2] & 0xff));
+
+			@SuppressWarnings("unchecked")
+			T result = (T) Character.valueOf(value);
+			return result;
+		}
+		case BOOLEAN: {
+			byte value = data[1];
+			@SuppressWarnings("unchecked")
+			T result = (T) (value == 0 ? Boolean.FALSE : Boolean.TRUE);
+			return result;
+		}
+		case FLOAT: {
+			int value = (
+					((data[1] & 0xff) << 24) +
+					((data[2] & 0xff) << 16) +
+					((data[3] & 0xff) << 8) +
+					(data[4] & 0xff));
+			@SuppressWarnings("unchecked")
+			T result = (T) Float.valueOf(Float.intBitsToFloat(value));
+			return result;
+		}
+		case DOUBLE: {
+			long value = (
+					((long) data[1] << 56) +
+					((long) (data[2] & 0xff) << 48) +
+					((long) (data[3] & 0xff) << 40) +
+					((long) (data[4] & 0xff) << 32) +
+					((long) (data[5] & 0xff) << 24) +
+					((long) (data[6] & 0xff) << 16) +
+					((long) (data[7] & 0xff) << 8) +
+					((long) (data[8] & 0xff) << 0));
+			@SuppressWarnings("unchecked")
+			T result = (T) Double.valueOf(Double.longBitsToDouble(value));
+			return result;
+		}
+		case BIGINTEGER: {
+			DataInputStream stream = new DataInputStream(new ByteArrayInputStream(data, 1, data.length - 1));
+			try {
+				String string = stream.readUTF();
+				@SuppressWarnings("unchecked")
+				T result = (T) new BigInteger(string);
+				return result;
+			} catch (IOException exception) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 469, 480, 620, 644.